pkg/commands/git_commands/sync.go

```diff
@@ -49,10 +49,14 @@ func (self *SyncCommands) Push(task gocui.Task, opts PushOpts) error {
 	return cmdObj.Run()
 }
 
+func (self *SyncCommands) fetchCommandBuilder(fetchAll bool) *GitCommandBuilder {
+	return NewGitCmd("fetch").
+		ArgIf(fetchAll, "--all").
+		ArgIf(!self.version.IsOlderThan(2, 29, 0), "--no-write-fetch-head")
```

Comment:
Could we add a comment explaining what this flag does and why we're using it?

Reply:
Thanks that's perfect. I started writing out a comment explaining my thoughts on code comments vs commit messages but I actually need to spend some more time thinking about it before getting back to you haha